src/ctapipe/core/tool.py

```diff
@@ -443,6 +449,19 @@ def run(self, argv=None, raises=False):
                 Provenance().finish_activity(
                     activity_name=self.name, status="interrupted", exit_code=exit_status
                 )
+            except self.custom_exceptions as err:
```

Comment:
Why do we need a custom exception here? Couldn't we just do:

```
exit_status =  getattr(err, "exit_code", 1):
```
in the existing exception path?

Reply:
To observe the difference is between "expected" (known, with specific exit code defined), and "unexpected" exceptions. If this is too narrow use case, we can follow your suggestion, but I'd prefer to distinguish between them.

Reply:
But the handling is the exact same

Reply:
I am really reluctant to add more complex logic to this already very complex tool that also has interactions with the traitlets base class.